packages/nextjs-mf/src/plugins/JsonpChunkLoading.ts

```diff
@@ -0,0 +1,314 @@
+// CustomWebpackPlugin.ts
+import type {
+  Chunk,
+  Compilation,
+  Compiler,
+  Module,
+  RuntimeModule,
+} from 'webpack';
+import { ConcatSource } from 'webpack-sources';
+// @ts-ignore
+import JsonpChunkLoadingRuntimeModule from 'webpack/lib/web/JsonpChunkLoadingRuntimeModule';
+import Template from '../../utils/Template';
+import { DEFAULT_SHARE_SCOPE_BROWSER } from '../internal';
+import { loadDependenciesTemplate } from './LoadDependenciesTemplate';
+
+// @ts-ignore
+export const setEagerLoading = (RuntimeGlobals) => {
+  return Template.asString([
+    'function setEagerLoading(libraryName, version, fromValue) {',
+    Template.indent([
+      'var library = ' +
+        RuntimeGlobals.shareScopeMap +
+        '.default[libraryName];',
+      'var shareScopeMap = ' + RuntimeGlobals.shareScopeMap + '.default;',
+      'var alternativeVersion = Object.keys(shareScopeMap[libraryName])[0];',
+      'if(!library) return;',
+      'var pickedVersion = library[version] || library[alternativeVersion];',
+      'if ((pickedVersion && pickedVersion.from === fromValue) && preferredModules.includes(libraryName)) {',
+      Template.indent([
+        "console.log('setting eager', libraryName, pickedVersion, fromValue);",
+        'pickedVersion.eager = true;',
+      ]),
+      '}',
+    ]),
+    '}',
+  ]);
+};
+
+function asyncOperationTemplate(
+  moduleArray: any[],
+  //@ts-ignore
+  RuntimeGlobals: any
+): Template {
+  return Template.asString([
+    `var initialConsumes = ${JSON.stringify(moduleArray)}`,
```

Comment:
Track the module requirements of all entrypoints. See what initial consume-shared-module types are within these entrypoints. 

These are considered eager calls and webpack must have these references resolved before the entry chunk executes. 

If you implement a import() in your page, the number of initialConsumes modules go down. As there's a async boundary that webpack can figure out and load those shared modules along with the feature chunk. 

InitialConsumes is also what webpack boot checks for when "shared module not available for eager consumption" if any of those maps factory calls return a promise not a function, it throws.